src/parser.yy

```diff
@@ -330,8 +330,9 @@ expr : int                                      { $$ = $1; }
      | expr DOT INT                             { $$ = new ast::FieldAccess($1, $3, @3); }
      | expr PTR ident                           { $$ = new ast::FieldAccess(new ast::Unop(token::MUL, $1, @2), $3, @$); }
      | expr "[" expr "]"                        { $$ = new ast::ArrayAccess($1, $3, @2 + @4); }
-     | "(" IDENT ")" expr %prec CAST            { $$ = new ast::Cast($2, false, $4, @1 + @3); }
-     | "(" IDENT MUL ")" expr %prec CAST        { $$ = new ast::Cast($2, true, $5, @1 + @4); }
+     | "(" IDENT ")" expr %prec CAST            { $$ = new ast::Cast($2, false, false, $4, @1 + @3); }
+     | "(" IDENT MUL ")" expr %prec CAST        { $$ = new ast::Cast($2, true, false, $5, @1 + @4); }
+     | "(" IDENT MUL MUL ")" expr %prec CAST    { $$ = new ast::Cast($2, true, true, $6, @1 + @5); }
```

Comment:
Wouldn't doing `ast::Cast(ast::Cast(...), ...)` be possible, that way we can easily do N depth pointers (although I never seen it going deeper than 3 so it might not be useful)

Reply:
I'm not sure that works. `ast::Cast(ast::Cast(...), ...)` is equivalent to `(uint8 *) (uint8 *) $foo` instead of `(uint8 **) $foo`, right?

Reply:
Hmm you're right. If we ever need more nesting we could do something like 


```
Cast::Cast(const std::string &type,
           bool is_pointer,
           int   depth,
           Expression *expr) {
....
   for (i < depth)
      cast.type = CreatePointer(cast.type)

```

Or even lift it to the CreatePointer method.

But I don't see a reason for this now

Reply:
Yeah, it should be fairly simple. I grepped around linux codebase and there's only a handful of triples pointers and no quadruple pointers.